### PR TITLE
Add Micrometer metrics option

### PR DESCRIPTION
**Pull request by checketts**
Addresses #185 by adding micrometer metrics.

One difference to the Yammer based approach is that operations and results are tags, which will allow for combining meters on the same chart easily.

### CODE REVIEW

File: folsom-micrometer-metrics/src/test/java/com/spotify/folsom/client/MicrometerMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    while (expectedValue != timer.count()) {
+      if (System.currentTimeMillis() - t1 > timeout) {
+        assertEquals(expectedValue, timer.count());
+        return;
+      }
+      Thread.sleep(0, 100);
+    }
+  }
+
+//        registry.gauge("memcache.outstandingRequests", tags, this, MicrometerMetrics::getOutstandingRequests);
```
Comment (protocol7) [inline folsom-micrometer-metrics/src/test/java/com/spotify/folsom/client/MicrometerMetricsTest.java]: Nit: remove commented code

File: folsom-micrometer-metrics/src/main/java/com/spotify/folsom/client/MicrometerMetrics.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    public MicrometerMetrics(final MeterRegistry registry, String... tags) {
+        this(registry, Tags.of(tags));
+    }
+
+    /**
+     *
+     * @param registry MeterRegistry
+     * @param tags Additional tags that will be added to each meter
+     */
+    public MicrometerMetrics(final MeterRegistry registry, Tags tags) {
```
Comment (protocol7) [inline folsom-micrometer-metrics/src/main/java/com/spotify/folsom/client/MicrometerMetrics.java]: Nit: be consistent with final for parameters and variables.

**Comment by protocol7** [comment]
Thanks, added a few nitpicks but looks great.

**Comment by checketts** [comment]
@protocol7 Thanks! I've pushed those changes.

**Review by protocol7 — APPROVED** [review]

**Comment by protocol7** [comment]
Thanks!

**Pull request merged**